client/trino-client/src/main/java/io/trino/client/StatementStats.java

```diff
@@ -194,12 +206,49 @@ public StageStats getRootStage()
     @JsonProperty
     public OptionalDouble getProgressPercentage()
     {
+        if (faultTolerantExecution) {
+            return getFaultTolerantExecutionSplitPercentage(true, false);
+        }
+
         if (!scheduled || totalSplits == 0) {
             return OptionalDouble.empty();
         }
         return OptionalDouble.of(min(100, (completedSplits * 100.0) / totalSplits));
     }
 
+    public OptionalDouble getFaultTolerantExecutionRunningPercentage()
+    {
+        return getFaultTolerantExecutionSplitPercentage(false, true);
+    }
+
+    private OptionalDouble getFaultTolerantExecutionSplitPercentage(boolean includeCompletedSplits, boolean includeRunningSplits)
```

Comment:
does this one guarantee that progress will not decrease? totalSplits can grow for stage, so I do not believe it is guaranteed. Am I missing something, or we just not care?

Reply:
Oh I didn't know `totalSplits` can grow per stage. But still, seems this is the best thing we can do now.